lib/scala/pkg/src/main/scala/org/enso/pkg/ComponentGroup.scala

```diff
@@ -0,0 +1,316 @@
+
+  /** [[Encoder]] instance for the [[Shortcut]]. */
+  implicit val encoder: Encoder[Shortcut] = { shortcut =>
+    shortcut.key.asJson
+  }
+
+  /** [[Decoder]] instance for the [[Shortcut]]. */
+  implicit val decoder: Decoder[Shortcut] = { json =>
+    ConfigCodecs.getScalar("shortcut", json).map(Shortcut(_))
+  }
+}
+
+/** The reference to a module.
+  *
+  * @param libraryName the qualified name of a library where the module is defined
+  * @param moduleName the module name
+  */
+case class ModuleReference(
+  libraryName: LibraryName,
+  moduleName: Option[ModuleName]
+)
+object ModuleReference {
+
+  private val ModuleSeparator: Char = '.'
+
+  private def toModuleString(moduleReference: ModuleReference): String = {
+    val libraryName =
+      s"${moduleReference.libraryName.namespace}$ModuleSeparator${moduleReference.libraryName.name}"
+    moduleReference.moduleName.fold(libraryName) { moduleName =>
+      s"$libraryName$ModuleSeparator${moduleName.name}"
+    }
+  }
+
+  /** [[Encoder]] instance for the [[ModuleReference]]. */
+  implicit val encoder: Encoder[ModuleReference] = { moduleReference =>
+    toModuleString(moduleReference).asJson
+  }
+
+  /** [[Decoder]] instance for the [[ModuleReference]]. */
+  implicit val decoder: Decoder[ModuleReference] = { json =>
+    json.as[String].flatMap { moduleString =>
+      moduleString.split(ModuleSeparator).toList match {
+        case namespace :: name :: module =>
+          Right(
+            ModuleReference(
+              LibraryName(namespace, name),
+              ModuleName.fromComponents(module)
+            )
+          )
+        case _ =>
+          Left(
+            DecodingFailure(
+              s"Failed to decode '$moduleString' as module reference. " +
+              s"Module reference should consist of a namespace (author), " +
+              s"library name and a module name (e.g. Standard.Base.Data).",
+              json.history
+            )
+          )
+      }
+    }
+  }
+
+}
+
+/** The module name.
+  *
+  * @param name the module name
+  */
+case class ModuleName(name: String)
+object ModuleName {
+
+  def fromComponents(items: List[String]): Option[ModuleName] =
+    Option.unless(items.isEmpty)(ModuleName(items.mkString(".")))
+
+  /** [[Encoder]] instance for the [[ModuleName]]. */
+  implicit val encoder: Encoder[ModuleName] = { moduleName =>
+    moduleName.name.asJson
+  }
+
+  /** [[Decoder]] instance for the [[ModuleName]]. */
+  implicit val decoder: Decoder[ModuleName] = { json =>
+    json.as[String] match {
+      case Left(_) =>
+        Left(
+          DecodingFailure(
+            "Failed to decode component group name.",
+            json.history
+          )
+        )
+      case Right(name) =>
+        Right(ModuleName(name))
+    }
+  }
+}
+
+/** The qualified library name.
+  *
+  * @param namespace the library namespace
+  * @param name the library name
+  */
+case class LibraryName(namespace: String, name: String)
```

Comment:
An exactly the same definition is already in the `editions` subproject on which `pkg` depends, so I don't see why it couldn't be reused.